### PR TITLE
Allow multiple TitleID for different versions for NSP files

### PR DESCRIPTION
**Pull request by garoxas**
lots of changes on different places, but mostly it's just because I changed all dictionaries from using TitleID as key to a pair (tuple) of TitleID and version as key, and unfortunately there's no simpler way to do this besides replacing all dictionary logic

other than that I've also:

- fixed some crashes while navigating the listview (dunno if this is regression from my changes or it's already there in the first place 😀 )
- NSP files patched to have RequiredSystemVersion value of 0 are now shown with Firmware 0 in the list
- fix DLC having empty game names caused by the app mistakenly use other function while trying to load scene XML

as always if you need more just hit me here on on temp ✌️

### CODE REVIEW

**Comment by garoxas** [comment]
I forgot to mention, I'll need to do this for XCI as well because there are some games with several versions (Zelda BOTW to name one), but it'll need to wait until at least this weekend so I can spare some time to do it

File: Switch Backup Manager/Form1.cs [hunk 1]
```diff
@@ -1235,14 +1236,14 @@ private void toolStripMenuItem16_Click(object sender, EventArgs e)
         {
             if (cbxRemoveableDrives.Items.Count > 0 && cbxRemoveableDrives.SelectedIndex > -1)
             {
-                Dictionary<Tuple<string, string>, FileData> list = DiffLists(SceneReleasesList, SDCardList);
+                Dictionary<Tuple<string, string>, FileData> list = DiffLists(SDCardList, SceneReleasesList);
```
Comment (garoxas) [inline Switch Backup Manager/Form1.cs]: yes this is important 😁

**Comment by gibaBR** [comment]
Hey, thank you! I'll nerd some time to proccess all those changes. I'm
working on getting extra info for titles from N's homepage. It Will be
optional as it will be slow because has to go to web for every title.

Em seg, 6 de ago de 2018 13:13, Garoxas <notifications@github.com> escreveu:

**Comment by garoxas** [comment]
Sure, take your time :+1:

**Comment by gibaBR** [comment]
I was thinking of something simplier, like we add a "_number" after the title ID (when more than one). Than, we could put that option to get the exact revison when scrapping as an option to the user, using a parameter on config form. 
And something else that I've been thinking is to allow user to edit data, so they could put manually set whatever they want. That would be done with a new form and user should right click the game they want to edit. What do you think?

**Comment by garoxas** [comment]
No, I mean like for example I have these 2 Zelda XCI down here
sbm
for now it's easy to figure out which one is the 1st revision and which is the latest revision because I still use the original scene release names. but if I rename them then there's no way for me to know which one is which  

If it's okay with you maybe I can post this on the temp so it's open to other members as well for them to share ideas  
for the mean time I'll try to find more ideas and if I can solve it then I'll make another PR separated from this one

as for the second one, well for sure user will be happier with more options right?  😄

**Comment by gibaBR** [comment]
Ok, got it. Nice idea to ask for help, maybe somebody can have a good idea to solve it.
I'll start to work on your PR this week.
Meanwhile, please take a look at this pre-release: 

https://mega.nz/#!SSxWgKzS!aTMutLiM-06mZTHoRS43B6omO4ytl9JphOL_AIaOq_0

Don't forget to check "scrap extra info from web" on config form.
Say what do you think of that. It can scrap info from e-shop for almost all NSP files. For XCI is not that good but I'm working on that.

**Comment by garoxas** [comment]
This is nice. Not as slow as I initially expected, tested with some XCIs and it works great, and on top of that this feature is optional so I'd say go for it

**Comment by gibaBR** [comment]
Just merged your commit and it breakes the NSP list... :)
On UpdateLocalNSPGamesList(), where you add a tag to the item. It seems to get the wrong version to the TitleID and then, on DisplayGameInfo() it cant find the object and we gert nullPointerException. I'm trying to fix that but maybe you could do it faster. Thanks!

**Comment by garoxas** [comment]
I couldn't reproduce the issue, but I encountered this one while I'm doing the multiple XCI revision, so I guess I can use the same fix I did there
as you might've noticed that I stored the version data using OSVListView.Tag, but apparently the Tag data is not carried over when you select an item so when you access ObjectListView.SelectedItems[0].Tag it'll return null data
so now I use the object data inside the OSVListView directly to get Version info
could you please try this one and see if it fixes the issue. thanks

**Comment by garoxas** [comment]
sorry for that 😁 
I've created another PR because I don't think I can modify this one anymore, so please check that one
thank you

**Pull request merged**